### ISSUE TITLE
formatdate ignores literal letters at the end of specification

### ISSUE THREAD

**Issue by voron** [hashicorp/terraform#26356]
Example from [docs](https://www.terraform.io/docs/configuration/functions/formatdate.html) doesn't work
### Terraform Version

```
terraform version
Terraform v0.13.3
```
### Expected Behavior
```
terraform console
> formatdate("H 'o''clock'", "2018-01-02T23:12:01-08:00")
11 o'clock
```
### Actual Behavior
```
terraform console
> formatdate("H 'o''clock'", "2018-01-02T23:12:01-08:00")
11 
> formatdate("H 'o''clock' H", "2018-01-02T23:12:01-08:00")
11 o'clock 11
```
### Steps to Reproduce
use `terraform console` to reproduce

**Comment by apparentlymart** [comment]
Thanks for reporting this, @voron. I was able to reproduce it in `terraform console`, as you suggested, in a development build of Terraform 0.14.

I've not yet done a deep investigation to fully explain the cause of this bug, but I suspect it's an issue in the upstream library that provides this date formatting functionality, possibly in [the internal function it uses to split the format string into directive vs. literal sequences](https://github.com/zclconf/go-cty/blob/36785d4dc4ac7d38d1a930390a805d23d63a99da/cty/function/stdlib/datetime.go#L344:6). Since it seems like an upstream bug I'm going to also mark it with our usual label to indicate that, although deeper investigation may lead in a different direction.

**Comment by apparentlymart** [comment]
Hi all!

This is now fixed in the main branch and in v0.14.0-beta1, due to the changes in #26585. That included @OwenTuz's zclconf/go-cty#69 which fixed the bug upstream. Thanks!.

```
$ terraform console
> formatdate("H 'o''clock'", "2018-01-02T23:12:01-08:00")
"11 o'clock"
```
